pyfreenas/pool.py

```diff
@@ -0,0 +1,115 @@
+from enum import Enum, unique
+from typing import TypeVar
+
+TMachine = TypeVar("TMachine", bound="Machine")
+TPoolStatus = TypeVar("TType", bound="PoolStatus")
+
+
+@unique
+class PoolStatus(Enum):
+    # States from "man zpool"
+    DEGRADED = "DEGRADED"
+    FAULTED = "FAULTED"
+    OFFLINE = "OFFLINE"
+    ONLINE = "ONLINE"
+    REMOVED = "REMOVED"
+    UNAVAIL = "UNAVAIL"
+    # State unique to FreeNAS (AFAIK)
+    # Pool known by FreeNAS, but not by zfs (I think)
+    UNKNOWN = "UNKNOWN"
+
+    @classmethod
+    def fromValue(cls, value: str) -> TPoolStatus:
+        if value == cls.DEGRADED.value:
+            return cls.DEGRADED
+        if value == cls.FAULTED.value:
+            return cls.FAULTED
+        if value == cls.OFFLINE.value:
+            return cls.OFFLINE
+        if value == cls.ONLINE.value:
+            return cls.ONLINE
+        if value == cls.REMOVED.value:
+            return cls.REMOVED
+        if value == cls.UNAVAIL.value:
+            return cls.UNAVAIL
+        if value == cls.UNKNOWN.value:
+            return cls.UNKNOWN
+        raise Exception(f"Unexpected pool status '{value}'")
+
+
+class Pool(object):
+    def __init__(self, machine: TMachine, guid: str) -> None:
+        self._machine = machine
+        self._guid = guid
+        self._cached_state = self._state
+
+    @property
+    def available(self) -> bool:
+        """If the pool exists on the Machine."""
+        return self._guid in self._machine._state["pools"]
+
+    @property
+    def encrypt(self) -> int:
+        """The encrypt? of the pool."""
+        if self.available:
+            self._cached_state = self._state
+            return self._state["encrypt"]
+        return self._cached_state["encrypt"]
+
+    @property
+    def guid(self) -> str:
+        """The guid of the pool."""
+        return self._guid
+
+    @property
+    def id(self) -> int:
+        """The id of the pool."""
+        if self.available:
+            self._cached_state = self._state
+            return self._state["id"]
+        return self._cached_state["id"]
+
+    @property
+    def is_decrypted(self) -> bool:
+        """Is the pool decrypted."""
```

Comment:
nit: either "is the pool decrypted?" or something like "Indicates if the pool decrypted or not."